autofs-5.1.6 - update sss timeout documentation

From: Ian Kent <raven@themaw.net>

Update then man page documentation and the configuration file
comments to reflect the changes to map reading and key lookups
with the sss lookup module.
---
 CHANGELOG                      |    1 +
 man/autofs.conf.5.in           |   20 +++++++++++++-------
 redhat/autofs.conf.default.in  |   28 +++++++++++++++++++---------
 samples/autofs.conf.default.in |   26 ++++++++++++++++++--------
 4 files changed, 51 insertions(+), 24 deletions(-)

diff --git a/CHANGELOG b/CHANGELOG
index 2d98a3a..e48e2c0 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -105,6 +105,7 @@ xx/xx/2018 autofs-5.1.5
 - sss introduce calculate_retry_count() function.
 - move readall into struct master.
 - sss introduce a flag to indicate map being read.
+- update sss timeout documentation.
 
 19/12/2017 autofs-5.1.4
 - fix spec file url.
diff --git a/man/autofs.conf.5.in b/man/autofs.conf.5.in
index 8dbc313..57dd421 100644
--- a/man/autofs.conf.5.in
+++ b/man/autofs.conf.5.in
@@ -31,9 +31,11 @@ overrides this and sets the timeout to 5 minutes to be consistent
 with earlier autofs releases.
 .TP
 .B master_wait
-sets the default maximum time to wait for the master map to become
-available if it cannot be read at program start (program default 10,
-wait for 10 seconds then continue).
+Sets the default maximum number of retries (actaul iterations is half
+this, each is delayed by 2 seconds before retrying) waiting for the
+master map to become available if it cannot be read at program start
+(program default 10, then continue). This can be longer if the map
+source itself waits for availability (such as sss).
 .TP
 .B negative_timeout
 .br
@@ -160,10 +162,14 @@ behaviour and so is an opt-in setting.
 .TP
 .B sss_master_map_wait
 .br
-Set the time to wait and retry if sssd returns "no such entry" when starting
-up. When sssd is starting up it can sometimes return "no such entry" for a
-short time until it has read in the LDAP map information. Default is 0 seconds,
-don't wait.
+Set the time to wait and retry if sssd is unable to read the master map
+at program start. Program default is 0 (don't wait) or 10 if sss supports
+returning EHSTDOWN when the provider isn't available.
+
+If the sss library supports returning EHOSTDOWN when the provider is down
+then this value is how long to wait between retries reading the master map.
+When reading dependent maps or looking up a map key this value is multiplied
+by the number of retries that would be used when reading the master map.
 .TP
 .B use_mount_request_log_id
 .br
diff --git a/redhat/autofs.conf.default.in b/redhat/autofs.conf.default.in
index bd52730..2a5a1c2 100644
--- a/redhat/autofs.conf.default.in
+++ b/redhat/autofs.conf.default.in
@@ -14,10 +14,13 @@
 #
 timeout = 300
 #
-# master_wait - set the default maximum time to wait for the
-# 		master map to become available if it cannot
-# 		be read at program start (default 10, wait
-# 		for 10 seconds then continue).
+# master_wait - set the default maximum number of retries (actual
+# 		iterations is half this, each is delayed by 2 seconds
+# 		before retrying) waiting for the master map to become
+# 		available if it cannot be read at program start
+# 		(default 10, then continue). This can be longer
+# 		if the map source itself waits for availability
+# 		(such as sss).
 #
 #master_wait = 10
 #
@@ -193,12 +196,19 @@ mount_nfs_default_protocol = 4
 #use_ignore_mount_option = no
 #
 # sss_master_map_wait - When sssd is starting up it can sometimes return
-# 			"no such entry" for a short time until it has read
+#			"no such entry" for a short time until it has read
 # 			in the LDAP map information. Internal default is 0
-# 			seconds, don't wait but if there is a problem with
-# 			autofs not finding the master map at startup (when
-# 			it should) then try setting this to 10 to work
-# 			around it.
+# 			(don't wait) or 10 if sss supports returning EHSTDOWN.
+# 			If there is a problem with autofs not finding the
+# 			master map at startup (when it should) then try setting
+# 			this to 10 or more. If the sss library supports returning
+# 			EHOSTDOWN when the provider is down then this value
+# 			is how long to wait between retries reading the
+# 			master map. When reading dependent maps or looking
+# 			up a map key this value is multiplied by the number
+# 			of retries that would be used when reading the master
+# 			map. (Default, 0 or 10 if sss suppprts returning
+# 			EHOSTDOWN).
 #
 #sss_master_map_wait = 0
 #
diff --git a/samples/autofs.conf.default.in b/samples/autofs.conf.default.in
index 1f55709..dd716c6 100644
--- a/samples/autofs.conf.default.in
+++ b/samples/autofs.conf.default.in
@@ -14,10 +14,13 @@
 #
 timeout = 300
 #
-# master_wait - set the default maximum time to wait for the
-# 		master map to become available if it cannot
-# 		be read at program start (default 10, wait
-# 		for 10 seconds then continue).
+# master_wait - set the default maximum number of retries (actual
+# 		iterations is half this, each is delayed by 2 seconds
+# 		before retrying) waiting for the master map to become
+# 		available if it cannot be read at program start
+# 		(default 10, then continue). This can be longer
+# 		if the map source itself waits for availability
+# 		(such as sss).
 #
 # master_wait = 10
 #
@@ -194,10 +197,17 @@ browse_mode = no
 # sss_master_map_wait - When sssd is starting up it can sometimes return
 #			"no such entry" for a short time until it has read
 # 			in the LDAP map information. Internal default is 0
-# 			seconds, don't wait but if there is a problem with
-# 			autofs not finding the master map at startup (when
-# 			it should) then try setting this to 10 to work
-# 			around it.
+# 			(don't wait) or 10 if sss supports returning EHSTDOWN.
+# 			If there is a problem with autofs not finding the
+# 			master map at startup (when it should) then try setting
+# 			this to 10 or more. If the sss library supports returning
+# 			EHOSTDOWN when the provider is down then this value
+# 			is how long to wait between retries reading the
+# 			master map. When reading dependent maps or looking
+# 			up a map key this value is multiplied by the number
+# 			of retries that would be used when reading the master
+# 			map. (Default, 0 or 10 if sss suppprts returning
+# 			EHOSTDOWN).
 #
 #sss_master_map_wait = 0
 #
